src/components/Fuel/FuelForm.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from 'react';
+import { Button, Label, Select, TextInput } from 'flowbite-react';
+import {
+  FuelSourceUnit,
+  FuelSourceType,
+  UnregisteredFuelRequest,
+} from '../../schema/fuel.schema';
+
+interface FuelFormProps {
+  handleSubmit: (fuelData: UnregisteredFuelRequest) => void;
+}
+
+export default function FuelForm({ handleSubmit }: FuelFormProps) {
+  /* Initializing state */
+  const [fuelData, setfuelData] = useState({
+    fuel_source_type: 'ng' as FuelSourceType,
+    fuel_source_unit: 'btu' as FuelSourceUnit,
+    fuel_source_value: 0 as number,
+  });
+
+  function submitForm(e: React.FormEvent) {
+    e.preventDefault();
+    console.log(fuelData);
+    handleSubmit(fuelData);
+  }
+
+  /* Function that updates the state */
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    /* Destructuring the event.target object */
+    const { name, value, valueAsNumber } = event?.target;
```

Comment:
Remove `valueAsNumber`. It won't be there nor do you need it. `parseFloat()` exists and that'll do the trick.